astropy/modeling/core.py

```diff
@@ -3341,6 +3341,81 @@ def render(self, out=None, coords=None):
         return out
 
 
+    def replace_submodel(self, name=None, model=None):
```

Comment:
When I read `replace_submodel`, I would tend to assume that it's modifying the same instance in place, eg. like assignment  to a list element. What's happening here seems more like `copy_with_new_submodel`, though that's a clumsier name. What do people think?

Reply:
I think that would definitely be the case if it was None-returning, so `m.replace_submodel()` would modify `m` in-place. This style of naming is like `str.replace(old_string, new_string)` so I think it's fine.